__tests__/crowdfundings/addPaymentSource.test.js

```diff
@@ -0,0 +1,104 @@
+const {
+  Cards,
+  createSource,
+  resetCustomers
+  } = require('./stripeHelpers')
+
+const PAY_PLEDGE_MUTATION = `
+  mutation addPaymentSource($sourceId: String!, $pspPayload: JSON!) {
+    addPaymentSource(sourceId: $sourceId, pspPayload: $pspPayload) {
+      isDefault
+      status
+      brand
+      expMonth
+      expYear
+    }
+  }
+`
+
+const addPaymentSource = async ({ sourceId, pspPayload }) => {
+  return apolloFetch({
+    query: PAY_PLEDGE_MUTATION,
+    variables: {
+      sourceId,
+      pspPayload
+    }
+  })
+}
+
+const prepare = async (options) => {
+  await connectIfNeeded()
+  await resetCustomers(pgDatabase())
+}
+
+test('addPaymentSource: adding a card as anonymous', async (t) => {
+  await prepare()
+  await signIn({ user: Users.Anonymous })
+  const source = await createSource({ card: Cards.Visa })
+  const result = await addPaymentSource({ sourceId: source.id, pspPayload: '' })
+  t.equal(result.errors[0].message, 'Sie müssen sich zuerst einloggen.', 'mutation fails because not logged in')
+  await signOut()
+  t.end()
+})
+
+test('addPaymentSource: adding a 3d secure card', async (t) => {
+  await prepare()
+  const { userId } = await signIn({ user: Users.Member })
+  const source = await createSource({ card: Cards.Visa3D, userId })
+  const result = await addPaymentSource({ sourceId: source.id, pspPayload: { type: 'three_d_secure' } })
+  t.equal(result.errors[0].message, 'Es tut uns leid, aus technischen Gründen können wir zurzeit kein Monatsabo auf eine Kreditkarte buchen, die 3D secure voraussetzt. Sie können es entweder mit einer anderen Karte versuchen und/oder uns kontaktieren unter: kontakt@republik.ch. ', 'mutation fails because 3d secure currently not allowed')
+  await signOut()
+  t.end()
+})
+
+test('addPaymentSource: adding an expired card', async (t) => {
+  await prepare()
+  const { userId } = await signIn({ user: Users.Member })
+  const source = await createSource({ card: Cards.Expired, userId })
+  const result = await addPaymentSource({ sourceId: source.id, pspPayload: {} })
+  t.equal(result.errors[0].message, 'Your card has expired.', 'mutation fails because card is expired')
+  await signOut()
+  t.end()
+})
+
+test('addPaymentSource: adding two cards', async (t) => {
+  await prepare()
+  const { userId } = await signIn({ user: Users.Member })
+  const untrustedSource = await createSource({ card: Cards.Untrusted, userId })
+  const untrustedResult = await addPaymentSource({ sourceId: untrustedSource.id, pspPayload: {} })
+  t.notOk(untrustedResult.errors, 'has no errors, untrusted card accepted')
+  t.deepEqual(untrustedResult.data, {
+    addPaymentSource: [{
+      isDefault: true,
+      status: 'CHARGEABLE',
+      brand: 'Visa',
+      expMonth: parseInt(Cards.Untrusted.exp_month, 10),
+      expYear: parseInt(Cards.Untrusted.exp_year, 10)
+    }]
+  }, 'added first card (untrusted, review), automatically became default card')
```

Comment:
please also check `user.paymentSources` to return the same as the mutation

Reply:
paymentSources table is empty for non-postfinance payments

Reply:
I see, but could you check it for `STRIPE`?

Reply:
<img width="805" alt="bildschirmfoto 2018-01-09 um 16 46 43" src="https://user-images.githubusercontent.com/2109393/34729354-b9a25a90-f55c-11e7-9f59-20a9fd6c3a42.png">

Reply:
I could test it for Postfinance Payments :)